File: 2021/j3.py
# code by savir singh

list1 = []
list2 = []
checks = []
counter = 0
previous = ""

while True:
    x = input()
    if x == "99999":
        break
    list1.append(x[2:])
    checks.append(x[:2])
    counter += 1
    if (int(x[0]) + int(x[1])) == 0:
        list2.append(previous)
    elif (int(x[0]) + int(x[1])) % 2 == 0:
           list2.append("right")
           previous = "right"
    else:
        list2.append("left")
        previous = "left"

try:
    if list1[0] != "99999":
        if checks[0] == "00":
            list2[0] = ""
            
except IndexError:
    pass


for i in range(counter):
    print(list2[i], list1[i])

File: 2021/j3.cpp
// code by savir singh

#include <bits/stdc++.h>
#include <iostream>
#include <string>
using namespace std;

int main() {

    string instruction, last3, prevdir;

    while (true) {
        last3 = "";
        cin >> instruction;

        last3.push_back(instruction[2]);
        last3.push_back(instruction[3]);
        last3.push_back(instruction[4]);

        int first, second, sum;
        first = instruction[0] - '0';
        second = instruction[1] - '0';
        sum = first+second;

        if (instruction == "99999") {
            break;
        }

        else if (sum%2==0 && sum != 0) {
            cout << "right " + last3 << endl;
            prevdir = "right ";
        }
        else if (sum == 0) {
            cout << prevdir + last3 << endl;
        }
        else if (sum%2!=0 && sum != 0) {
            cout << "left " + last3 << endl;
            prevdir = "left ";
        }
    }

}
